### DISCUSSION TITLE
Playwright tests in TS is much slower than JS

### DISCUSSION THREAD

**Discussion by shiva-taramani** [artilleryio/artillery#3242] [Artillery]
Hi, I would like some feedback and directions.

I used transpiled JS for my Artillery Playwright tests, which worked like a charm. I was excited about the experimental TypeScript support and tried using that. My tests are much slower to the point that I am back to transpiling my code again.

Any suggestions or ideas would be much appreciated.

Thank you!

**Comment by hassy** [comment]
Very interesting & unexpected! We may need some extra information to try to figure out what's going on. Artillery uses [esbuild](https://esbuild.github.io/) under the hood to transpile TypeScript. How are you transpiling your TypeScript code? /cc @bernardobridge

**Reply by shiva-taramani** [reply]
Just tsc like ```"test:perf:survey": "tsc && npm run build:scenarios && npm run test:artillery:survey",```

**Reply by shiva-taramani** [reply]
I also wonder if it had to do with my code creating a browser for every vuser. I will try with typescript again to see if that was it.

**Reply by shiva-taramani** [reply]
Yes, this has to do with the code creating browser for each vuser.

**Reply by hassy** [reply]
😅 okay good to know!
